### ISSUE TITLE
Required m2o in o2m update fails validation

### ISSUE THREAD

**Issue by Thomasparsley** [directus/directus#5070]
**1) The issue and what you expected to happen**
In product collection is M2O relation. I am logged as admin user.
If i want to create new M2O or edit existing M2O i got this error:
![image](https://user-images.githubusercontent.com/41927757/114837067-96e23700-9dd3-11eb-870f-5306f4ab5fce.png)

When I open existing item collection i cant see which user create M2O item.
![image](https://user-images.githubusercontent.com/41927757/114836707-3a7f1780-9dd3-11eb-96d8-72dd4e927e2f.png)

**2) Exact steps to reproduce this issue**

1. Create `product` collection
2. Create `sub_product` collection
3. In `sub_product` collection add M2O (`sub_product` -> `product`)
4. Create new `product` with `sub_products`

**3) Your environment:**
Directus 9.0.0-rc. ~~58 59~~ 60
DBMS: MariaDB ~~10.3.25~~ 10.5.9
Deployment: npx
Browser: Chrome ~~89~~ 90 - 64x
Node: ~~14.16.0~~ 15.14.0.
~~Local VM - Ubuntu 20.10~~ Windows 10

**4) Any other relevant information we might need to reproduce this issue**
[mariadb_dump.zip](https://github.com/directus/directus/files/6316349/greys_dump.zip)

**Comment by skoontastic** [comment]
Seeing the same thing with rc 59 when I try to create a "many" relationship from the "one" side. So in my case I have a collection called Schools, and schools can have many Departments. If I'm on a School and try to create a Department I get the same validation error as the OP. It seems like the field for the relationship just isn't getting updated.

![CleanShot 2021-04-16 at 11 19 14](https://user-images.githubusercontent.com/585102/115054230-9f0aa700-9ea5-11eb-89f0-40327155e5bd.png)

**Comment by Thomasparsley** [comment]
> Seeing the same thing with rc 59

I can confirm.

**Comment by rijkvanzanten** [comment]
This might be a duplicate of https://github.com/directus/directus/issues/4245 🤔  @skoontastic Could you check if you happen to have 2 rows in `directus_relations` about the same relationship instead of 1?

**Comment by skoontastic** [comment]
Looks like I do?

```
sqlite> SELECT * FROM directus_relations;
1|departments|school|id|schools|departments|id||||
2|programs|department|id|departments|programs|id||||
3|alerts|user_created|id|directus_users||id||||
4|alerts|user_updated|id|directus_users||id||||
5|calendars|user_created|id|directus_users||id||||
6|calendars|user_updated|id|directus_users||id||||
7|calendars_calendarable|calendars_id|id|calendars|calendarable|id|||item|
8|calendars_calendarable|item|id||||collection|departments,programs,schools|calendars_id|
9|events|user_created|id|directus_users||id||||
10|events|user_updated|id|directus_users||id||||
11|events|calendar|id|calendars||id||||
12|events_eventable|events_id|id|events|eventable|id|||item|
13|events_eventable|item|id||||collection|calendars,departments,programs,schools|events_id|
14|departments_directus_users|departments_id|id|departments|users|id|||directus_users_id|
15|departments_directus_users|directus_users_id|id|directus_users|departments|id|||departments_id|
16|posts|user_created|id|directus_users||id||||
17|posts|user_updated|id|directus_users||id||||
18|posts_postable|posts_id|id|posts||id|||item|
19|posts_postable|item|id||||collection|departments,programs,schools|posts_id|
20|posts_schools|posts_id|id|posts|schools|id|||schools_id|
21|posts_schools|schools_id|id|schools|posts|id|||posts_id|
22|posts_departments|departments_id|id|departments|posts|id|||posts_id|
23|posts_departments|posts_id|id|posts|departments|id|||departments_id|
24|navigation_menus_related_pages|navigation_menus_id|id|navigation_menus|related_pages|id|||item|sort
25|navigation_menus_related_pages|item|id||||collection|departments,posts,programs,schools,external_links|navigation_menus_id|
26|posts_programs|posts_id|id|posts|programs|id|||programs_id|
27|posts_programs|programs_id|id|programs|posts|id|||posts_id|
```

**Comment by rijkvanzanten** [comment]
Hmm i'm not sure. As per that related ticket I would've expected to see `departments` in `many_collection` twice with the same `many_field`, but that doesn't seem to be the case here.

**Comment by Thomasparsley** [comment]
> When I open existing item collection i cant see which user create M2O item.

When the APP loads the M2O and obtains the user's uuid, the APP does not request the necessary user data. Simplified order of HTTP API requests:
1. Fetch `store_product` with ID 6.
2. Fetch all relationships for `store_product`

But when there is a user uuid in the relationships, the APP does not request user data

> If i want to create new M2O or edit existing M2O i got this error:

I read the entire console logs with `DEBUG = knex: *`. I think the transaction to create new / update data will go fine. According to the debug from Knex.

But I wonder why in this query:
![image](https://user-images.githubusercontent.com/41927757/115138153-c3dc5700-a02a-11eb-845b-336426f5fc3c.png)
is
```
`store_product_nutrition`.`id` not in (?,?))
```
When binding, ID 9 is an existing record and ID 11 is a new record ...

When i try execute this SQL i got empty return.
```
select `store_product_nutrition`.`id`
from `store_product_nutrition`
where (`store_product_nutrition`.`product` = 6 and `store_product_nutrition`.`id` not in (9, 11))
order by `store_product_nutrition`.`id` asc limit 100
```

**Comment by Thomasparsley** [comment]
Interesting... When I allow create new record for public access(for all collections in relationship), then validateJoi don't throw any error.
https://github.com/directus/directus/blob/d9b0fc1e339419a9ae4fd954b9e8a6edc4b3bdd8/api/src/services/authorization.ts#L257-L261

rc.60

**Comment by rijkvanzanten** [comment]
This is caused by the fact that the API tries to nullify all the deselected items in the o2m. This fails (correctly), as `null` isn't allowed as per the required state (allow null = false). 

We're about to implement support for delete-on-deselect (instead of the default nullify on deselect) in https://github.com/directus/directus/issues/2942, which will also solve this issue.

Closed in favor of https://github.com/directus/directus/issues/2942

**Comment by fballan93** [comment]
is this fixed in rc 62 ? because i can still see the error even after update

**Comment by rijkvanzanten** [comment]
@fballan93 Read my last message above in which I closed this issue:

> We're about to implement support for delete-on-deselect (instead of the default nullify on deselect) in #2942, which will also solve this issue.
> Closed in favor of #2942

This issue was closed, as the implementation of #2942 will also be the solution for this problem. #2942 hasn't been implemented yet, so this issue hasn't been fixed either
